### PR TITLE
Add basic map to test movement and weapons

### LINKED ISSUE #357: Training mode
Arena where you can pick any gun part and shoot at dummies

- [ ] Map (reuse CraterTown assets)
- [ ] Platforms that let you choose guns? Alt some set of stuff
- [ ] Requires level/mode select

### PR DESCRIPTION
**Pull request by Fueredoriku**
# Movement
- [x] Three boxes to showcase ideal Leap height, jump height and leapdash height
- [x] Three ranges of blocks that showcase the length of each jump type (not beyond first leapdash)
- [x] The two fan types with pillars that showcase how high they push the player

# Interactable props
- [x] Explosive barrels, showing how far they reach with one barrel inside and another outside the range
- [x] Soda cans

# Shooting
- [x] #88 (or rather a punching bag, didn't implement a standing dummy yet)
- ~~Item select screen beforehand??? (not yet, methinks)~~
- [x] Or a bunch of interactables that switch out your barrel at least? (may want to disable some functionality when not in editor) (may need extensive testing)

# Menu integration
- [x] Make it possible to exit back to main menu (through an exit gate since we ain't got no pause menu)
- [x] ~~New sub-menu for main menu after start with "level select" and "training" button? Perhaps new entry in the main menu?~~ New (barebones) level select sub menu after player select.

# Miscellaneous
- Spawn points now rotate players properly, closing #395 
- Sun size is adjusted to better match how it looks on Mars (source: NASA)
- Interact keybind that raycasts to check for interactables (only used for exit door atm)
- StaticInfo now has a field for starting extension, closing #396 

Closes #357 

Current progress:  
image

### CODE REVIEW

File: Assets/Scripts/Gamestate/PlayerIdentity.cs [hunk 1]
```diff
@@ -98,11 +98,12 @@ public void resetItems()
         Extensions = new List<Item>();
         body = StaticInfo.Singleton.StartingBody;
         barrel = StaticInfo.Singleton.StartingBarrel;
-        extension = null;
+        extension = StaticInfo.Singleton.StartingExtension;
```
Comment (Fueredoriku) [inline Assets/Scripts/Gamestate/PlayerIdentity.cs:101]: Yeeess

**Review by Fueredoriku — COMMENTED** [review]
Great work! I really love the damage numbers and such

Seems sodie popper is missing from the item selection. We could consider populating the stands with augments found in staticInfo automatically, but that might be overkill for this PR

You can self-approve at you own discretion 👍

**Comment by toberge** [comment]
> You can self-approve at you own discretion 👍

You may want to have another look as I made a ton of changes just now with fixes to various other bugs that appeared when swapping to some specific augments.

File: Assets/Scripts/Interactables/TrainingModeAugmentPlacer.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    [SerializeField]
+    private Vector3 columnSpace = 4 * Vector3.left;
+
+    [SerializeField]
+    private Vector3 rowSpace = 2 * Vector3.forward;
+
+    [SerializeField]
+    private TrainingModeAugment pedestal;
+
+    // Start is called before the first frame update
```
Comment (Fueredoriku) [inline Assets/Scripts/Interactables/TrainingModeAugmentPlacer.cs:15]: Maybe remove default comment

**Review by Fueredoriku — COMMENTED** [review]
If you fix this error it should be ready to merge 👍
image

File: Assets/Scripts/Interactables/TrainingModeAugmentPlacer.cs (hunk 2 above)
Comment (toberge) [reply Assets/Scripts/Interactables/TrainingModeAugmentPlacer.cs]: Done!

**Review by toberge — APPROVED** [review]
Fredrik said it's ready & I fixed the things

**Pull request merged**